Dockerfile

```diff
@@ -1,26 +1,90 @@
-from alpine as build-environment
+# syntax=docker/dockerfile:1.4
+FROM alpine:3.15 AS build-environment
 WORKDIR /opt
-RUN apk add clang lld curl build-base linux-headers git \
-    && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh \
-    && chmod +x ./rustup.sh \
-    && ./rustup.sh -y
+
+RUN apk update
+
 WORKDIR /opt/foundry
+
+RUN set -eux; \
+	\
+	apk add --no-cache --virtual .foundry-deps \
+		ca-certificates \
+		clang \
+		lld \
+        build-base \
+        linux-headers \
+        git \
+        curl \
+        findutils \
+	; \
+	\
+	dpkgArch="$(dpkg --print-architecture | awk -F- '{ print $NF }')"; \
+	curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh; \
+    chmod +x ./rustup.sh; \
+    ./rustup.sh -y; \
+    export CARGO_HOME=/opt/foundry/.cargo;
+
+ENV CARGO_HOME=/opt/foundry/.cargo
+
+# Copy seperate
+COPY ./Cargo.toml /opt/foundry/Cargo.toml
+COPY ./Cargo.lock /opt/foundry/Cargo.lock
 COPY . .
-RUN source $HOME/.profile && cargo build --release \
-    && strip /opt/foundry/target/release/forge \
-    && strip /opt/foundry/target/release/cast \
-    && strip /opt/foundry/target/release/anvil
 
-from alpine as foundry-client
+RUN source $HOME/.profile && cargo build --release
+
+# build for release
+RUN rm -rf /opt/foundry/target/release/deps/
+RUN rm -rf /opt/foundry/target/release/build/
```

Comment:
I don't think this is needed given we only copy the target binaries? Actually, this might even slow down the build since it can't do incremental compilation? :thinking: